### DISCUSSION TITLE
Saving observation states and actions at the end of each episode

### DISCUSSION THREAD

**Discussion by dhruvkm2402** [Toni-SM/skrl#46] [General]
Hello,
I've implemented the tool and it's really good w.r.t flexibility and one code file which is so much easier. Thank you for that.
I wanted to know if we can save the states and actions at the end of each episode?, as that would be a useful data later.

**Comment by Toni-SM** [comment]
Hi @dhruvkm2402 

It is possible to save, in persistent storage, all environment transitions (state, action, next state, reward, terminated, and truncated) and any other information stored in memory automatically when the memory is filled

Supported formats are PyTorch (`.pt`), NumPy (`.np`), and Comma-separated values (`.csv`)

Please visit the [Memory export/import](https://skrl.readthedocs.io/en/latest/intro/data.html#memory-export-import) section

**Reply by dhruvkm2402** [reply]
Thank you for pointing it out
